extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php

```diff
@@ -201,87 +195,118 @@ private function getUploadWarningMessages( $warnings ){
 		return $ret;
 	}
 
-	public function executeWordmarkUpload() {
-		// check rights
-		if ( !ThemeDesignerHelper::checkAccess() ) {
-			$this->displayRestrictionError( __METHOD__ );
-		}
+	public function wordmarkUpload() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		// SUS-797: Validate edit token and POST request for external requests
+		$this->checkWriteRequest();
 
 		$upload = new UploadWordmarkFromFile();
 
 		$status = $this->uploadImage($upload);
 
 		if($status['status'] === 'uploadattempted' && $status['isGood']) {
 			$file = $upload->getLocalFile();
-			$this->wordmarkImageUrl = wfReplaceImageServer( $file->getUrl() );
-			$this->wordmarkImageName = $file->getName();
+			$wordmarkImageUrl = wfReplaceImageServer( $file->getUrl() );
+			$wordmarkImageName = $file->getName();
 
 			// if wordmark url is not set then it means there was some problem
-			if ( $this->wordmarkImageUrl == null || $this->wordmarkImageName == null ) {
-				$this->errors = array( wfMsg( 'themedesigner-unknown-error' ) );
+			if ( $wordmarkImageUrl == null || $wordmarkImageName == null ) {
+				$this->response->setData( [
+					'errors' => [ wfMessage( 'themedesigner-unknown-error' )->escaped() ]
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "errors" 13 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "themedesigner-unknown-error" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)